api/src/main/java/io/strimzi/api/kafka/model/AbstractSsLike.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.fabric8.kubernetes.api.model.Affinity;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.KubeLink;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A baseclass for {@link Kafka} and {@link Zookeeper} with all their common properties.
+ */
+@Buildable(editableEnabled = false,
+        generateBuilderPackage = true,
+        builderPackage = "io.strimzi.api.kafka.model"
+)
+public abstract class AbstractSsLike {
```

Comment:
In the work I'd doing for the `KafkaConnect` kind I've reused this class again, and since the connect support isn't SS based I renamed it to `ReplicatedJvmPods`. It's a bit of a weak abstraction, to be honest, just serving to group together some properties, and saving us from having the declare them on the (now 3) different subclasses.

Reply:
So why not just `AbstractAssembly` ? Why the `Jvm' in the name ? Replicated pods sounds really weird to me.

Reply:
Because in the `KafkaAssembly` (the Java representation of a `kind: Kafka` resource) case it's the superclass of the `Kafka` and `Zookeeper`, which are not assemblies.

But that gives me the idea that `KafkaAssembly` should really be called `KafkaKind`, which is both shorter and more more obvious. Then I would have `KafkaKindOperator`, which resolves a naming conflict with the existing `KafkaAssemblyOperator`. 

Anyway, I thought 'jvm needed to be in there since one of the properties is `jvmOptions`, and 'replicated' because this class also has `replicas`. Really the rubbish name is a reflection of the weak abstraction this class provides: its a base class for things from which we'll generate things which have replicated pods (SS or Deployment) which are running Java.